lib/trino-parquet/src/main/java/io/trino/parquet/reader/flat/FlatColumnReader.java

```diff
@@ -68,11 +70,16 @@
 
     private int nextBatchSize;
 
-    public FlatColumnReader(PrimitiveField field, ValueDecodersProvider<BufferType> decodersProvider, ColumnAdapter<BufferType> columnAdapter)
+    public FlatColumnReader(
+            PrimitiveField field,
+            ValueDecodersProvider<BufferType> decodersProvider,
+            ColumnAdapter<BufferType> columnAdapter,
+            LocalMemoryContext memoryContext)
     {
         this.field = requireNonNull(field, "field is null");
         this.decodersProvider = requireNonNull(decodersProvider, "decoders is null");
         this.columnAdapter = requireNonNull(columnAdapter, "columnAdapter is null");
+        this.memoryContext = requireNonNull(memoryContext, "memoryContext is null");
```

Comment:
should you account for retained memory too? Is it significant?

Reply:
So far I haven't found the sizes of individual objects to be significant enough to matter. Could be later if we find that it actually matters.
Note that even though ORC reader has code to calculate retained size of objects, it also does not try to account them in the memory usage logic.